app/src/main/kotlin/org/eurofurence/connavigator/ui/fragments/UserStatusFragment.kt

```diff
@@ -36,6 +43,20 @@ class UserStatusFragment : DisposingFragment(), AnkoLogger {
 
         // Subscribe to changes in PM availability.
         subscribeToPMs()
+
+        // Subscriber to database changes
+        subscribeToDatabase()
+    }
+
+    private fun subscribeToDatabase() {
+        subscriptions += db.subscribe {
+            val state = it.state
+            val canLogin = state != null && state.toLowerCase() != "past"
+
+            ui.apply {
+                layout.visibility = if (canLogin) View.VISIBLE else View.GONE
+            }
```

Comment:
The class is subtype of `DisposingFragment`, `.collectOnDestroyView()` should be called to collect the database subscription on view destruction. The code now just collects the subscription disposable but does not actually dispose of them. (The code in `UserStatusFragment` apparently has the same mistake.)